core/src/test/java/org/opensearch/sql/planner/optimizer/LogicalPlanOptimizerTest.java

```diff
@@ -305,6 +311,52 @@ public PhysicalPlan implement(LogicalPlan plan) {
         () -> table.createWriteBuilder(null));
   }
 
+  @Test
+  void paged_table_scan_builder_support_project_push_down_can_apply_its_rule() {
+    when(tableScanBuilder.pushDownProject(any())).thenReturn(true);
+    when(table.createPagedScanBuilder(anyInt())).thenReturn(tableScanBuilder);
+
+    var relation = new LogicalRelation("schema", table);
+    relation.setPageSize(anyInt());
+
+    assertEquals(
+        tableScanBuilder,
+        LogicalPlanOptimizer.paginationCreate().optimize(project(relation))
+    );
+  }
+
+  @Test
+  void push_page_size() {
+    var relation = new LogicalRelation("schema", table);
+    var paginate = new LogicalPaginate(42, List.of(project(relation)));
```

Comment:
I've noticed a magic number 42 is used a few times throughout these files. Does this need to be a constant?